sympy/polys/rings.py

```diff
@@ -2531,6 +2531,124 @@ def compose(f, x, a=None):
 
         return poly
 
+
+    def coeff_wrt(self, sym, dg):
+        """
+        Coefficient of ``f`` with respect to ``x**deg``.
+
+        Treating ``f`` as a univariate polynomial in ``x`` this finds the
+        coefficient of ``x**deg`` as a polynomial in the other generators.
+
+        Parameters
+        ==========
+
+        p : sympy.Poly
+            The polynomial to compute the sparse coefficient for.
+        sym : int
+            The symbol to compute the coefficient for.
+        dg : int
+            The degree of the monomial to compute the coefficient for.
+
+        Returns
+        =======
+
+        sympy.Number
+            The sparse coefficient of the polynomial at the given symbol and degree.
+
+        Examples
+        ========
+
+        >>> from sympy.polys import ring, ZZ
+        >>> _, x, y, z = ring("x, y, z", ZZ)
+        >>> p = 2*x**4 + 3*y**4 + 10*z**2 + 3
+        >>> dg = 2
+        >>> sym = 2
+        >>> p.coeff_wrt(sym, dg)
+        10
+
+        """
+        p = self
+        terms = [(m, c) for m, c in p.iterterms() if m[sym] == dg]
+        monoms, coeffs = zip(*terms)
+        monoms = [m[:sym] + (0,) + m[sym + 1:] for m in monoms]
+        return p.ring.from_dict(dict(zip(monoms, coeffs)))
+
+
+    def prem2(self, g, x):
+        """
+        Computes the pseudo-remainder of the polynomial `f` with respect to `g`.
+
+        Parameters
+        ==========
+
+        f : PolyElement
+            The polynomial to compute the pseudo-remainder for.
+        g : PolyElement
+            The polynomial to divide `f` by.
+        x : Symbol
+            The main variable of the polynomials.
```

Comment:
The `x` parameter here cannot be a Symbol so this is inaccurate.

Reply:
This should clarify that the `x` parameter is optional and explain what its default value represents.

Reply:
I'm unresolving this because it is not resolved:

The parameter is not a Symbol:
```
In [4]: import sympy

In [5]: x1 = sympy.Symbol('x')

In [6]: R, x2 = sympy.ring('x', ZZ)

In [7]: x1 == x2
Out[7]: False

In [8]: type(x1)
Out[8]: sympy.core.symbol.Symbol

In [9]: type(x2)
Out[9]: sympy.polys.rings.PolyElement
```
The parameter is either an `int` or a generator given as a `PolyElement` but not a `Symbol`. I would refer to it as a generator here and then use the doctest examples to show precisely what types it can be.

If the parameter is not provided then the default is that the first generator is chosen. That should be stated explicitly and also demonstrated in the doctest examples.

Reply:
Again I am unresolving this. If this parameter is optional then there needs to be an explicit explanation here of what happens when an argument is not supplied.